### PR TITLE
Update REST API calls that do not get a response object to no longer …

### PR DESCRIPTION
**Pull request by mikenevi**
…expect one.

### CODE REVIEW

File: MHVLib/Clients/MHVRemoteMonitoringClient.m [hunk 1]
```diff
@@ -48,13 +48,12 @@ - (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
     return self;
 }
 
-- (void)requestWithPath:(NSString *_Nonnull)path
+- (void)requestWithPathAndResultClass:(NSString *_Nonnull)path
```
Comment (miburf) [inline MHVLib/Clients/MHVRemoteMonitoringClient.m]: I wouldn't include the AndResultClass here, just the resultClass change below

File: MHVLib/Clients/MHVRemoteMonitoringClient.m [hunk 2]
```diff
@@ -104,4 +102,53 @@ - (void)requestWithPath:(NSString *_Nonnull)path
...
+    {
+        NSMutableString *queryPath = [path mutableCopy];
+        
+        [pathParams enumerateKeysAndObjectsUsingBlock:^(id _Nonnull key, id _Nonnull obj, BOOL *_Nonnull stop)
+         {
+             [queryPath replaceCharactersInRange:[queryPath rangeOfString:[NSString stringWithFormat:@"{%@}", key]] withString:obj];
+         }];
+        
+        path = queryPath;
+    }
```
Comment (miburf) [inline MHVLib/Clients/MHVRemoteMonitoringClient.m]: Both methods here do this, I'd make it into a updatePath:withPathParams: method they can share

Comment (miburf) [inline MHVLib/Clients/MHVRemoteMonitoringClient.m]: Both methods here do this, I'd make it into a updatePath:withPathParams: method they can share.

File: MHVLib/Models/Generated/Categories/MHVActionPlanTasksApi.h [hunk 3]
```diff
@@ -43,11 +43,10 @@ NS_ASSUME_NONNULL_BEGIN
 ///
 /// @param actionPlanTask 
 /// 
-///  code:200 message:"OK",
 ///  code:201 message:"Created",
 ///  code:0 message:"Error"
 ///
-/// @return NSObject*
+/// @return MHVActionPlanTaskInstanceV2*
```
Comment (miburf) [inline MHVLib/Models/Generated/Categories/MHVActionPlanTasksApi.h:49]: I know it's the generated code from that big template, but they really return void & have the completion that returns the result object or error.

**Review by miburf — APPROVED** [review]

File: MHVLib/Models/Generated/Categories/MHVActionPlanTasksApi.h (hunk 3 above)
Comment (mikenevi) [reply MHVLib/Models/Generated/Categories/MHVActionPlanTasksApi.h:49]: You're right, I'll make a note of that and maybe fix it later. It is generated code.

**Pull request merged**